stackslib/src/net/api/getblockbyheight.rs

```diff
@@ -0,0 +1,231 @@
+            ));
+        }
+
+        let block_height_str = captures
+            .name("block_height")
+            .ok_or_else(|| {
+                Error::DecodeError("Failed to match path to block height group".to_string())
+            })?
+            .as_str();
+
+        let block_height = block_height_str.parse::<u64>().map_err(|_| {
+            Error::DecodeError("Invalid path: unparseable block height".to_string())
+        })?;
+        self.block_height = Some(block_height);
+
+        Ok(HttpRequestContents::new().query_string(query))
+    }
+}
+
+impl RPCRequestHandler for RPCNakamotoBlockByHeightRequestHandler {
+    /// Reset internal state
+    fn restart(&mut self) {
+        self.block_height = None;
+    }
+
+    /// Make the response
+    fn try_handle_request(
+        &mut self,
+        preamble: HttpRequestPreamble,
+        contents: HttpRequestContents,
+        node: &mut StacksNodeState,
+    ) -> Result<(HttpResponsePreamble, HttpResponseContents), NetError> {
+        let block_height = self
+            .block_height
+            .take()
+            .ok_or(NetError::SendError("Missing `block_height`".into()))?;
+
+        let tip = match node.load_stacks_chain_tip(&preamble, &contents) {
+            Ok(tip) => tip,
+            Err(error_resp) => {
+                return error_resp.try_into_contents().map_err(NetError::from);
+            }
+        };
+
+        let index_block_hash_res =
+            node.with_node_state(|_network, _sortdb, chainstate, _mempool, _rpc_args| {
+                chainstate
+                    .index_conn()
+                    .get_ancestor_block_hash(block_height, &tip)
+            });
+
+        let block_id = match index_block_hash_res {
+            Ok(index_block_hash_opt) => match index_block_hash_opt {
+                Some(index_block_hash) => index_block_hash,
+                None => {
+                    // block hash not found
+                    let msg = format!("No such block #{:?}\n", block_height);
+                    warn!("{}", &msg);
+                    return StacksHttpResponse::new_error(&preamble, &HttpNotFound::new(msg))
+                        .try_into_contents()
+                        .map_err(NetError::from);
+                }
+            },
+            Err(e) => {
+                // error querying the db
+                let msg = format!("Failed to load block #{}: {:?}\n", block_height, &e);
+                warn!("{}", &msg);
+                return StacksHttpResponse::new_error(&preamble, &HttpServerError::new(msg))
+                    .try_into_contents()
+                    .map_err(NetError::from);
+            }
+        };
+
+        let stream_res =
+            node.with_node_state(|_network, _sortdb, chainstate, _mempool, _rpc_args| {
+                let Some((tenure_id, parent_block_id)) = chainstate
+                    .nakamoto_blocks_db()
+                    .get_tenure_and_parent_block_id(&block_id)?
```

Comment:
I don't think you want this -- this gets you the tenure ID and block ID of the parent of the given block (which doesn't point to the block the user is requesting).  I think you want `get_block_header_nakamoto_tenure_id()` so you can get the consensus hash for the block ID you just looked up.  You'd then feed that and the block ID into `NakamotoBlockStream::new()`.

Reply:
this is from block_v3 where it returns the parent id too, do you prefer i create a new NakamotoBlockStream struct?

Reply:
D'oh! You're right. What you have here is fine :+1: